node/src/main/scala/coop/rchain/node/configuration/Configuration.scala

```diff
@@ -39,87 +35,68 @@ object Configuration {
   private val KeyConfigKey         = s"${hocon.Tls.Key}.${hocon.Tls.keys.Key}"
   private val GenesisPathConfigKey = s"${hocon.Casper.Key}.${hocon.Casper.keys.GenesisPath}"
 
-  def build(arguments: Seq[String])(implicit log: Log[Task]): Task[Configuration] =
-    for {
-      config <- buildConfiguration(arguments).attempt
-      exit <- config.fold(
-               t => log.error(s"Can't build the configuration: ${t.getMessage}").as(true),
-               kp(Task.now(false))
-             )
-      _ = if (exit) System.exit(1)
-    } yield config.right.get
-
-  private def buildConfiguration(
-      arguments: Seq[String]
-  )(implicit log: Log[Task]): Task[Configuration] =
-    for {
-      options       <- Task.delay(commandline.Options(arguments))
-      profile       = options.profile.toOption.flatMap(profiles.get).getOrElse(defaultProfile)
-      command       = subcommand(options)
-      configRef     <- Task.delay(ConfigFactory.load())
-      config        <- createConfig(options, command, profile)
-      configuration = createConfiguration(config.withFallback(configRef), command)
-      _             = System.setProperty("rnode.data.dir", configuration.server.dataDir.toString)
-      _             <- if (command == Run) log.info(s"Starting with profile ${profile.name}") else Task.unit
-    } yield configuration.copy(printHelpToConsole = () => options.printHelp())
-
-  private def getDataDir(
-      options: commandline.Options,
-      command: Command,
-      profile: Profile
-  ): Path =
-    if (command == Run) options.run.dataDir.getOrElse(profile.dataDir._1())
-    else profile.dataDir._1()
-
-  private def createConfig(options: commandline.Options, command: Command, profile: Profile)(
-      implicit log: Log[Task]
-  ): Task[Config] =
-    for {
-      dataDir        <- Task.delay(getDataDir(options, command, profile))
-      baseConfig     = createBaseConfig(dataDir)
-      externalConfig <- createExternalConfig(dataDir, options, command)
-    } yield externalConfig.withFallback(baseConfig)
-
-  private def createBaseConfig(dataDir: Path): Config =
-    ConfigFactory.parseMap(
-      Map(DataDirConfigKey -> dataDir.toString).asJava,
-      "base configuration"
-    )
+  def build(arguments: Seq[String]): Configuration = {
+    val configurationE = Try(buildConfiguration(arguments)).toEither
 
-  private def createExternalConfig(dataDir: Path, options: commandline.Options, command: Command)(
-      implicit log: Log[Task]
-  ): Task[Config] =
-    createFileConfig(command, options.configFile.getOrElse(dataDir.resolve("rnode.conf")))
-      .map(ConfigMapper.fromOptions(options).withFallback(_))
+    configurationE match {
+      case Left(t) =>
+        System.err.println(s"Can't build the configuration: ${t.getMessage}")
+        System.exit(1)
+      case _ =>
```

Comment:
Why not return configuration here

case Right(configuration) => configuration.get

Reply:
@KentShikama Because `System.exit()` is of type `Unit`. The type of this function is `Config`. So the code wouldn't compile.